### PR TITLE
Add e5-mistral embedding model - step 3/3

### PR DESCRIPTION
**Pull request by Ying1123**
This is the last PR contains unit test after
- #987 
- #983 

Note: does not support input is a list of prompts

### CODE REVIEW

File: python/sglang/srt/model_executor/model_runner.py [hunk 1]
```diff
@@ -161,6 +161,10 @@ def load_model(self):
             vllm_model_config.hf_config.num_key_value_heads = 8
             monkey_patch_vllm_qvk_linear_loader()
 
+        if "e5-mistral" in self.server_args.model_path.lower():
```
Comment (Ying1123) [inline python/sglang/srt/model_executor/model_runner.py]: do not hardcode model name

File: python/sglang/srt/utils.py [hunk 2]
```diff
@@ -223,6 +223,13 @@ def is_multimodal_model(model):
     raise ValueError("unrecognized type")
 
 
+def is_embedding_model(model_path):
+    # FIXME incomplete list
+    if "e5-mistral-7b-instruct" in model_path.lower():
```
Comment (Ying1123) [inline python/sglang/srt/utils.py]: relax the constraint if possible

File: python/sglang/srt/managers/tp_worker.py [hunk 3]
```diff
@@ -596,15 +630,19 @@ def forward_decode_batch(self, batch: ScheduleBatch):
...
         output_rids = []
-        output_vids = []
-        decoded_texts = []
-        output_read_ids = []
-        output_read_offsets = []
-        output_skip_special_tokens = []
-        output_spaces_between_special_tokens = []
         output_meta_info = []
         output_finished_reason: List[BaseFinishReason] = []
+        if batch.is_embedding:
```
Comment (Ying1123) [inline python/sglang/srt/managers/tp_worker.py]: reverse the order of if/else branch

**Comment by merrymercy** [comment]
This PR introduces too many if/else. Follow this principle to refactor them
1. If there is a if/else branch, make sure the generation branch is the first one
2. If possible, always use `if is_generation` instead of `if not is_embedding`

File: python/sglang/srt/managers/tokenizer_manager.py [hunk 4]
```diff
@@ -459,19 +487,27 @@ def create_handle_loop(self):
...
-                out_dict = {
-                    "text": recv_obj.output_strs[i],
-                    "meta_info": recv_obj.meta_info[i],
-                }
+                if isinstance(recv_obj, BatchStrOut):
+                    out_dict = {
+                        "text": recv_obj.output_strs[i],
+                        "meta_info": recv_obj.meta_info[i],
+                    }
+                else:  # isinstance(recv_obj, BatchEmbeddingOut)
```
Comment (merrymercy) [inline python/sglang/srt/managers/tokenizer_manager.py]: `assert isinstance(recv_obj, BatchEmbeddingOut)`

**Review by merrymercy — CHANGES_REQUESTED** [review]

**Pull request merged**